build.sbt

```diff
@@ -277,4 +277,18 @@ lazy val core = (project in file("modules/core"))
     )
   )
 
+lazy val dag = (project in file("modules/dag"))
+  .dependsOn(core)
```

Comment:
If dag needs to depend on core it should be separated (see examples, not as a module). IMO the dependency should be in the opposite direction (core depends on dag)

Reply:
I think that @TheMMaciek is using `core` as an `sdk`. So the dependency is in the right direction (`dag` on `sdk`).

On the second issue of keeping it separate (like `examples`). I agree in principle, however `dag` is a special case of a state-channel, so maybe it is okay if we treat it specially and keep it as a module? What do you guys think?

Reply:
Maybe I could create something like `dag-core` module (probably dependent on schema only) and keep base types and validators there (stuff that's needed for snapshot creation in L0 also). And I would keep state channel specific implementation in `dag` module dependent on `core` or separate project? Regarding where to keep DAG state channel implementation dependent on `core` I think keeping it in a module would be nice, given that it is ours, it is closely related and kind of a backbone of the whole system.  And yes I am using the core as an sdk for the needed cluster functionality - I guess most of the state channels will probably do that. BTW that's why I think we shouldn't necessarily keep the stuff related to L0 in `core`. I guess `core` should be a base for both L1's and L0. WDYT?

Reply:
For me, `core` is the app that will be running on every L0 node. That's where all the base things should be, as well as DAG (maybe as a dependency). The other way would be to keep `core` as an sdk, and create module `l0` and separate project `l1`.